tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/QueryNegativeTests.java

```diff
@@ -69,8 +69,12 @@ public void testFromClauseWithInvalidType() {
         validateError(compileResult, index++, "redeclared symbol 'age'", 381, 44);
         validateError(compileResult, index++, "invalid constraint type. expected subtype of 'map<any|error>' " +
                 "but found 'int'", 401, 22);
-        validateError(compileResult, index, "invalid constraint type. expected subtype of 'map<any|error>' " +
+        validateError(compileResult, index++, "invalid constraint type. expected subtype of 'map<any|error>' " +
                 "but found 'int'", 411, 22);
+        validateError(compileResult, index++, "incompatible types: expected 'int[]', found '(int[]|other)'",
```

Comment:
We try to avoid error messages with "other" since it doesn't quite add any value to the user and almost always the error will change when the user fixes the original error.

In this scenario, the error seems to be logged because we seem to be creating a union type with symTable.semanticError as a member, which IMO shouldn't happen. Should we maybe add a validation to check for errors before creating this type and return symTable.semanticError if there is an error?

Reply:
Fixed the issue.